RSDEV-427 Analytics events for the new gallery

rlamacraft opened the pull request.

rlamacraft committed:
Page load event

rlamacraft committed:
Sidebar toggle event

rlamacraft committed:
Section change event

rlamacraft committed:
Change view event

rlamacraft committed:
Change sort event

rlamacraft committed:
Search

rlamacraft committed:
Tapped breadcrumb

rlamacraft committed:
Created folder

rlamacraft committed:
Uploaded file

rlamacraft committed:
Drag-and-drop files into Gallery

rlamacraft commented:
@tilorspace Before I do too many, any feedback on the naming of the events?

tilorspace commented:
@rlamacraft I haven't looked at all events, but user:uploaded:file:gallery for example looks good to me using actor:verb:object:target; there was however another one user:created:gallery:folder, which accordingly should be switched to user:created:folder:gallery. 
another thing, I'd suggest to use present tense for the verbs, e.g. upload, create, change, ...

rlamacraft committed:
mobile info panel opened

rlamacraft committed:
drags files

rlamacraft committed:
edit description

rlamacraft committed:
linked documents global id tapped

rlamacraft committed:
edit images

rlamacraft committed:
edit documents

rlamacraft committed:
more editing documents

rlamacraft committed:
Remove unused private property `#open`

This should have been removed by commit ea6fbe25, when it replaced the
mechanism for opening folders.

rlamacraft committed:
duplicate files

rlamacraft committed:
move files

rlamacraft committed:
rename file

rlamacraft committed:
Only update name after rename is successful

If the rename operation fails then we do not want to update the locally stored
name otherwise we end up in a corrupted local state.

rlamacraft committed:
upload new version

rlamacraft committed:
download

rlamacraft committed:
export

rlamacraft committed:
irods
